test/org/sagebionetworks/bridge/services/ReportServiceTest.java

```diff
@@ -0,0 +1,306 @@
+        ReportData someData = createReport(LocalDate.parse("2015-02-10"), "First", "Name");
+        
+        service.saveStudyReport(TEST_STUDY, IDENTIFIER, someData);
+        
+        verify(mockReportDataDao).saveReportData(reportDataCaptor.capture());
+        ReportData retrieved = reportDataCaptor.getValue();
+        assertEquals(someData, retrieved);
+        assertEquals(STUDY_REPORT_DATA_KEY.getKeyString(), retrieved.getKey());
+        assertEquals(LocalDate.parse("2015-02-10"), retrieved.getDate());
+        assertEquals("First", retrieved.getData().get("field1").asText());
+        assertEquals("Name", retrieved.getData().get("field2").asText());
+    }
+    
+    @Test
+    public void saveParticipantReportData() {
+        ReportData someData = createReport(LocalDate.parse("2015-02-10"), "First", "Name");
+        
+        service.saveParticipantReport(TEST_STUDY, IDENTIFIER, HEALTH_CODE, someData);
+        
+        verify(mockReportDataDao).saveReportData(reportDataCaptor.capture());
+        ReportData retrieved = reportDataCaptor.getValue();
+        assertEquals(someData, retrieved);
+        assertEquals(PARTICIPANT_REPORT_DATA_KEY.getKeyString(), retrieved.getKey());
+        assertEquals(LocalDate.parse("2015-02-10"), retrieved.getDate());
+        assertEquals("First", retrieved.getData().get("field1").asText());
+        assertEquals("Name", retrieved.getData().get("field2").asText());
+    }
+    
+    @Test
+    public void deleteStudyReport() {
+        service.deleteStudyReport(TEST_STUDY, IDENTIFIER);
+        
+        verify(mockReportDataDao).deleteReportData(STUDY_REPORT_DATA_KEY);
+    }
+    
+    @Test
+    public void deleteParticipantReport() {
+        service.deleteParticipantReport(TEST_STUDY, IDENTIFIER, HEALTH_CODE);
+        
+        verify(mockReportDataDao).deleteReportData(PARTICIPANT_REPORT_DATA_KEY);
+    }
+    
+    // The following are date range tests from MPowerVisualizationService, they should work with this service too
+    
+    @Test
+    public void defaultStartAndEndDates() {
+        // mock now
+        DateTimeUtils.setCurrentMillisFixed(DateTime.parse("2016-02-08T09:00-0800").getMillis());
+        try {
+            service.getParticipantReport(TEST_STUDY, IDENTIFIER, HEALTH_CODE, null, null);
+            
+            verify(mockReportDataDao).getReportData(PARTICIPANT_REPORT_DATA_KEY, LocalDate.parse("2016-02-07"), LocalDate.parse("2016-02-07"));
+        } finally {
+            DateTimeUtils.setCurrentMillisSystem();
+        }
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void startDateAfterEndDate() {
+        service.getParticipantReport(TEST_STUDY, IDENTIFIER, HEALTH_CODE, END_DATE, START_DATE);
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void dateRangeTooWide() {
+        service.getParticipantReport(TEST_STUDY, IDENTIFIER, HEALTH_CODE, START_DATE, START_DATE.plusDays(46));
+    }
```

Comment:
Should we also have the date validation tests and default date tests for Study reports as well?